src/redisraft.c

```diff
@@ -597,6 +621,11 @@ static bool handleInterceptedCommands(RedisRaftCtx *rr,
         return true;
     }
 
+    if (len == strlen("UNWATCH") && strncasecmp(cmd_str, "UNWATCH", len) == 0) {
```

Comment:
Why do we handle `unwatch` here but handle `watch` in the state machine? Let's send `unwatch` there. We can handle it next to `watch`.

Reply:
confused what you mean that UNWATCH isn't being handled in the state machine?  this appends a raft entry.  no actually unwatch happens here.  The point of handling it here, is to not handle it a "redis command" and append it as a redis command, but as a unique "end session" object.

Reply:
Currently, in `RaftExecuteCommandArray()` you have

```c
if (strncasecmp("watch", cmd, cmdlen) == 0) {
....
    createSession();
...
} 
```

you can add this part:

```c
if (strncasecmp("watch", cmd, cmdlen) == 0) {
...
    createSession();
...
} else if (strncasecmp("unwatch", cmd, cmdlen) == 0) {
...
    endSession();
...
}
```

Then, we don't need to intercept `unwatch` here (on append time), similar to `watch`. So, we handle these commands specially in the state machine. 

Then, we'll use `RAFT_LOGTYPE_END_SESSION` entry just for disconnect events.

Reply:
its not just for disconnect, it can also be used for multi abort / discard.

Reply:
We can directly write `UNWATCH`, `DISCARD` commands to the log and process differently (by destroying related session) in the state machine. 

It'll be same functionality-wise but a step closer to the "real sessions". Send commands/events to the state machine and process there.

Then, for the disconnect event, we create that entry(`RAFT_LOGTYPE_CLIENT_DISCONNECT`)  and write to the log. So, these two commands (unwatch, discard) and disconnect event will do the same thing currently: destroy the session. In future, maybe they won't. (depends on what we implement).

Reply:
the reason I didn't do it that way, is I dislike the text parsing of UNWATCH/DISCARD at apply time.  It just bothers me when dealing with the state machine to work based on string comparisons.  if we want to do it that way, I can implement it, just rus me the wrong way.

Reply:
ignoring for now as this is not critical